api/v1/ytsaurus_webhook.go

```diff
@@ -451,7 +463,37 @@ func (r *Ytsaurus) validateInstanceSpec(instanceSpec InstanceSpec, path *field.P
 	return allErrors
 }
 
-func (r *Ytsaurus) validateYtsaurus(old *Ytsaurus) field.ErrorList {
+func (r *Ytsaurus) validateExistsYTsaurus(ctx context.Context, ytsaurus *Ytsaurus) field.ErrorList {
+	var allErrors field.ErrorList
+
+	// if ytsaurus is nil, there no such object in the given namespace, we can proceed
+	if ytsaurus == nil {
+		return allErrors
+	}
+
+	cfg, err := config.GetConfig()
+	if err != nil {
+		allErrors = append(allErrors, field.InternalError(field.NewPath("config"), err))
+		return allErrors
+	}
+	k8sClient, err := client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		allErrors = append(allErrors, field.InternalError(field.NewPath("k8s client"), err))
+		return allErrors
+	}
+
+	var ytsaurusList YtsaurusList
+
+	err = k8sClient.List(ctx, &ytsaurusList, &client.ListOptions{Namespace: ytsaurus.Namespace})
```

Comment:
Also as far as I remember this k8sClient gives an error and all point of introducing custom validator is to use client which is came from the manager here.

Reply:
I suggest start by commenting other validators and make your new test with new validator (and kubeclient from validator struct) green. So we don't discover k8s call still don't work after fixing other validators.